### ISSUE TITLE
sxiv unable to open animated gif without file extension

### ISSUE THREAD

**Issue by eylles** [nsxiv/nsxiv#37]
tracking of [#412](https://github.com/muennich/sxiv/issues/412) on here, after some testing i was able to determine it is a problem with libgif and the way the support for animated gifs is implemented, as it seems to check for that animated gifs have the gif extension, didn't determine where exactly the check happens but i think the way to go is to check by "magic" (file type from header) than by extension.

**Comment by N-R-K** [comment]
> i was able to determine it is a problem with libgif

Are you sure? We don't even get to any libgif related functions from what I can tell. The problem lies here:
https://github.com/nsxiv/nsxiv/blob/c6a4320187a197cac014d6a5d07e003127f4637e/image.c#L304

**Comment by N-R-K** [comment]
Yeah, `imlib2` is the problem here. It cannot deal with extensionless gif files (neither can it deal with extensionless webp files).

We will have to mirror what we're doing on https://github.com/nsxiv/nsxiv/pull/20 and roll our own `is_gif` function to detect gif files. And if it is gif, bypass imlib and load it with giflib instead.

**Comment by N-R-K** [comment]
I've opened a [bug report](https://phab.enlightenment.org/T8957) about this for imlib2. I think we should try to get imlib2 to properly deal with extensionless files rather than dealing with it on our side.

**Comment by N-R-K** [comment]
> This problem was just fixed in git (yesterday). It was a regression since 1.6.x.
> Imlib2 uses the file name extension only as a first guess. After that it tries all loaders.
> All loaders do signature checks like you describe.

[- kwo](https://phab.enlightenment.org/T8957#155990)

**Comment by eylles** [comment]
nothing more we can do ¯\\\_(ツ)\_/¯
